services/search/pkg/search/service.go

```diff
@@ -323,6 +324,20 @@ func (s *Service) searchIndex(ctx context.Context, req *searchsvc.SearchRequest,
 			s.logger.Error().Err(err).Str("space", space.Id.OpaqueId).Str("mountpointId", mountpointID).Msg("invalid mountpoint space id")
 			return nil, errSkipSpace
 		}
+		// excludeq the hidden shares
+		rs, err := gatewayClient.GetReceivedShare(ctx, &collaborationv1beta1.GetReceivedShareRequest{
+			Ref: &collaborationv1beta1.ShareReference{
+				Spec: &collaborationv1beta1.ShareReference_Id{
+					Id: &collaborationv1beta1.ShareId{
+						OpaqueId: oid,
+					},
+				},
+			},
+		})
+		if err == nil && rs.GetStatus().GetCode() == rpcv1beta1.Code_CODE_OK && rs.GetShare().GetHidden() {
```

Comment:
Shouldn't we log something `if err != nil`? Or if we get an unexpected status code?

Reply:
:+1: Now you can get rid of the  `err == nil` requirement. It's checked before.

Do we want to check for `rs.GetStatus().GetCode() != rpcv1beta1.Code_CODE_OK` also? We dont really expect that....